packages/client/lib/sync/fetcher/blockfetcher.ts

```diff
@@ -24,13 +24,17 @@ export class BlockFetcher extends BlockFetcherBase<Block[], Block> {
   async request(job: Job<JobTask, Block[], Block>): Promise<Block[]> {
     const { task, peer } = job
     const { first, count } = task
-    const headers = await (peer!.eth as EthProtocolMethods).getBlockHeaders({
-      block: first,
-      max: count,
-    })
-    const bodies: BlockBodyBuffer[] = <BlockBodyBuffer[]>(
-      await peer!.eth!.getBlockBodies(headers.map((h) => h.hash()))
-    )
+    const headers = (
```

Comment:
```suggestion
       const headersResponse = await (peer!.eth as EthProtocolMethods).getBlockHeaders({
      block: first,
      max: count,
    })
    /* eslint-disable @typescript-eslint/no-unnecessary-condition*/
    // Conditional catches occasional null responses from peers who do not return block headers from peer.eth request
    if (!headersResponse) {
      this.config.logger.warn(
        `peer ${peer?.id} returned no headers for segment starting with block ${first}`
      )
      return []
    }
    /* eslint-enable  @typescript-eslint/no-unnecessary-condition*/
    const headers = headersResponse[1]
```

Reply:
This works.  I didn't realize the `getBlockHeaders` actually returned an array so the error was trying to access the headers from that array when a peer returns a null response.  This produces the intended result of an early return and logging a warning saying what happened (though maybe this should be a debug thing since the client will just fetch them from elsewhere if it doesn't get anything)?

Reply:
ah thanks, I think warn is fine for this